### DISCUSSION TITLE
print suppression

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#9408] [General]
Looking for a minimal effort path to turn off a ton of diagnostic prints (with strings & comma separated variables in them) I opted to bulk edit every occurrence of  print(  to  _prnt(  where  _prnt(  is
```
def _prnt(*args):
  if verbose: print(*args)
```
thinking I could use the verbose flag to switch the prints on & off at will. This worked right up to the first print with a , end='  ' in it, where it fell over with 
```
TypeError: unexpected keyword argument end
```

Can someone suggest a solution please? I've googled this to a dead end.

**Comment by robert-hh** [comment]
You could make end=None a keyword argument of your function, and then proceed depending on the value of end and verbose. like
Another way I have seen for that task is not to use this _prnt() function, but to write all print lines as:

`verbose and print(....)`

If verbose evaluates to False, the print statement will not be executed.

**Comment by cwalther** [comment]
How about
```python
def _prnt(*args, **kwargs):
    if verbose: print(*args, **kwargs)
```

(`*args` only catches positional arguments)

**Reply by kjm1102** [reply]
Tnx Chris, 3 out of 3 hat trick
```
def _prnt(*args, **kwargs): verbose and print(*args, **kwargs)

verbose=1
_prnt('text',2,3,end='  e')
_prnt()
_prnt('this' if verbose else 'that')

>>> 
text 2 3  e
this
>
```
